### PR TITLE
Implement barrierDismissible for `showCupertinoDialog`

### LINKED ISSUE #45744: `showCupertinoDialog` should support `barrierDismissible`
Internal: https://b.corp.google.com/issues/145094605

In `showCupertinoDialog` it is not supported to enabled to dismiss the dialog on outside tap. It seems that this might be the normal iOS behaviour, but I couldn't find any documentation that it is not allowed to support it.

### PR DESCRIPTION
**Pull request by tottomotto**
## Description

Adds support for dismissing cupertino dialog with barrier behavior. By default the dialog is not dismissible with a tap on the barrier. The localization files updates are based on the existing material ones.

## Related Issues

Fixes #45744

## Tests

I added the following tests:

- check if barrierDismissible is disabled by default
- check if barrierDismissible is enabled if enabled explicitly
- check for existence of the localized modalBarrierDismissLabel for cupertino

## Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

### CODE REVIEW

**Comment by Piinks** [comment]
Hi @tottomotto welcome! 🎉  Thank you for the contribution.
It looks like there are merge conflicts with this PR. Can you resolve them and update your branch?

File: packages/flutter/test/cupertino/dialog_test.dart [hunk 1]
```diff
@@ -52,6 +52,67 @@ void main() {
     expect(find.text('Delete'), findsNothing);
   });
 
+  testWidgets('Barrier dismissible', (WidgetTester tester) async {
```
Comment (xster) [inline packages/flutter/test/cupertino/dialog_test.dart]: Use complete sentences. Also if there are 2 things being tested, prefer adding 2 tests for 'Not barrier dismissible by default' and 'Configurable to be barrier dismissible'.

File: packages/flutter_localizations/lib/src/l10n/cupertino_en.arb [hunk 2]
```diff
@@ -5,82 +5,72 @@
...
   "copyButtonLabel": "Copy",
   "@copyButtonLabel": {
     "description": "The label for copy buttons and menu items. The reference abbreviation is what iOS shows on text selection toolbars."
   },
-
   "pasteButtonLabel": "Paste",
   "@pasteButtonLabel": {
     "description": "The label for paste buttons and menu items. The reference abbreviation is what iOS shows on text selection toolbars."
   },
-
```
Comment (xster) [inline packages/flutter_localizations/lib/src/l10n/cupertino_en.arb]: you can leave the formatting of this file. Super nit: leave the extra newline at the end of the file.

File: packages/flutter_localizations/lib/src/l10n/cupertino_en.arb [hunk 3]
```diff
@@ -5,82 +5,72 @@
...
   "selectAllButtonLabel": "Select All",
   "@selectAllButtonLabel": {
     "description": "The label for select-all buttons and menu items. The reference abbreviation is what iOS shows on text selection toolbars."
+  },
+  "modalBarrierDismissLabel": "Dismiss",
+  "@modalBarrierDismissLabel": {
+    "description": "Label read out by accessibility tools (VoiceOver) for a modal barrier to indicate that a tap dismisses the barrier. A modal barrier can for example be found behind an alert or popup to block user interaction with elements behind it."
```
Comment (xster) [inline packages/flutter_localizations/lib/src/l10n/cupertino_en.arb]: can, for example, be

Comment (xster) [reply packages/flutter_localizations/lib/src/l10n/cupertino_en.arb]: nit: wrap line at 100 characters.

File: packages/flutter_localizations/lib/src/l10n/cupertino_en_AU.arb [hunk 4]
```diff
@@ -18,5 +18,6 @@
   "cutButtonLabel": "Cut",
   "copyButtonLabel": "Copy",
   "pasteButtonLabel": "Paste",
-  "selectAllButtonLabel": "Select all"
-}
+  "selectAllButtonLabel": "Select all",
+  "modalBarrierDismissLabel": "Dismiss"
+}
```
Comment (xster) [inline packages/flutter_localizations/lib/src/l10n/cupertino_en_AU.arb]: leave the final newline in these files

**Review by xster — COMMENTED** [review]
Thanks for your contribution! 
LGTM minus a few nits. 

heads up @HansMuller in case you run the translation console anytime soon, otherwise I'll run it after this goes in

File: packages/flutter_localizations/lib/src/l10n/cupertino_en.arb (hunk 3 above)
Comment (tottomotto) [reply packages/flutter_localizations/lib/src/l10n/cupertino_en.arb]: I'm not sure what you mean by wrap :)

File: packages/flutter/test/cupertino/dialog_test.dart [hunk 5]
```diff
@@ -1129,3 +1185,16 @@ Widget boilerplate(Widget child) {
     child: child,
   );
 }
+
+Widget createAppWithCenteredButton(Widget child) {
+  return MaterialApp(
+      home: Material(
```
Comment (xster) [inline packages/flutter/test/cupertino/dialog_test.dart]: Be consistent with the rest of the file on indents (2 spaces)

**Review by xster — APPROVED** [review]

**Comment by xster** [comment]
some tests are failing (but not due to any issues in your PR). Could you rebase your PR on head and push again?

**Pull request merged**